src/marqo/tensor_search/tensor_search.py

```diff
@@ -2648,3 +2657,33 @@ def delete_documents(config: Config, index_name: str, doc_ids: List[str]):
             document_ids=doc_ids,
         )
     )
+
+def normalize_vector(embeddings: Union[List[List[float]], ndarray]) -> List[List[float]]:
+    """
+    Normalizes a list of vectors (embeddings) to have unit length.
+
+    Args:
+        embeddings (Union[List[List[float]], ndarray]): A list of vectors or a numpy ndarray of vectors to be normalized.
+
+    Returns:
+        List[List[float]]: A list of normalized vectors.
+    """
+
+    # Convert the input embeddings to a numpy array
+    if embeddings.__class__ == ndarray:
```

Comment:
Why do we need to handle ndarray type?

Reply:
we're using this function in 3 places, out of which 1 passes a ndarray (here: https://github.com/marqo-ai/marqo/blob/c0d09826a6c662fb96501f3e9f429cde94ba6a18/src/marqo/tensor_search/tensor_search.py#L1920) 

`merged_vector = np.mean(weighted_vectors, axis=0)` will convert it to an ndarray, which will then be passed to normalize_vector where we don't need to convert it to ndarray again

Reply:
this is where the multimodal combo field embedding is calculated. According to Owen and Jesse, although it's quite rare, but it is possible that the norm is 0. So it's also good to call normalize_vector method here. Thanks for fixing this.